nose_launchable/plugin.py

```diff
@@ -52,30 +49,42 @@ def configure(self, options, conf):
         super(Launchable, self).configure(options, conf)
 
         self.subset_enabled = options.subset_enabled or False
+        self.record_only_enabled = options.record_only_enabled or False
+
         self.build_number = options.build_number or os.getenv(BUILD_NUMBER_KEY)
         self.subset_target = options.subset_target
-
         self.subset_options = options.subset_options
 
-        self.record_only_enabled = options.record_only_enabled or False
-
         if not (self.subset_enabled or self.record_only_enabled):
+            # we didn't get activated
+            return
+
+        if os.environ[LaunchableClientFactory.TOKEN_KEY] is None:
+            logger.warning("%s not set. Deactivating"%LaunchableClientFactory.TOKEN_KEY)
             return
 
         if self.subset_enabled and self.record_only_enabled:
-            self.enabled = False
             logger.warning("Please specify either --launchable-subset or --launchable-record-only flag")
             return
 
-        self.enabled = True
-
-        if self.enabled and self.build_number is None:
-            self.enabled = False
+        if self.build_number is None:
             logger.warning("Please specify --launchable-build-number flag")
+            return
 
-        if self.subset_enabled and not ((self.subset_target is None) ^ (self.subset_options is None)):
-            self.enabled = False
+        if not ((self.subset_target is None) ^ (self.subset_options is None)):
             logger.warning("Please specify either --launchable-subset-target or --launchable-subset-options flag")
+            return
```

Comment:
I found the previous logic somewhat confusing. Namely at some arbitrary point `self.enabled` is set to `True` and every error check afterward would have to turn it back off if it finds a problem.

I believe more consistent way to do this is to do all error checks first, then only after everything checks out enable this plugin.